integration-libs/punchout/components/punchout-requisition/punchout-requisition.component.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  inject,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { RoutingService } from '@spartacus/core';
+import {
+  PUNCHOUT_ERROR_PAGE_URL,
+  PunchoutFacade,
+  PunchoutRequisition,
+} from '@spartacus/punchout/root';
+import { filter, map, Observable, switchMap, take, tap, timer } from 'rxjs';
+
+@Component({
+  selector: 'cx-punchout-requsition',
+  templateUrl: './punchout-requisition.component.html',
+  changeDetection: ChangeDetectionStrategy.Default,
+  standalone: false,
+})
+export class PunchoutRequisitionComponent implements OnInit {
+  @ViewChild('punchoutFormElement')
+  punchoutFormElement!: ElementRef<HTMLFormElement>;
+  protected punchoutFacade = inject(PunchoutFacade);
+  protected routingService = inject(RoutingService);
+  punchoutFormGroup: FormGroup;
+
+  protected formBuilder = inject(FormBuilder);
+
+  punchoutRequisition$: Observable<PunchoutRequisition | undefined> =
+    this.punchoutFacade.getPunchoutSessionRequisition().pipe(
+      take(1),
+      tap((punchoutRequisition) => {
+        this.listenAndSubmitForm(punchoutRequisition).subscribe({
+          next: () => {
+            this.punchoutFormElement.nativeElement.submit();
+          },
+          error: () => {
+            this.routingService.go(PUNCHOUT_ERROR_PAGE_URL);
+          },
+        });
+        this.punchoutFormGroup.setValue({
+          order: punchoutRequisition?.orderAsCXML,
+        });
+      })
+    );
+
+  ngOnInit(): void {
+    this.punchoutFormGroup = this.formBuilder.group({
+      order: [''],
+    });
+  }
+
+  protected listenAndSubmitForm(
+    req: PunchoutRequisition | undefined
+  ): Observable<boolean> {
+    return this.punchoutFormGroup.controls['order'].valueChanges.pipe(
```

Comment:
Consider using strongly typed forms with `TypedFormGroup` and `TypedFormBuilder` for better type safety.
Maybe we could use a const like this:
```
const FORM_CONTROL_NAME = {
  ORDER: 'order',
} as const;
```

```
this.punchoutFormGroup = this.formBuilder.group({
   [FORM_CONTROL_NAME.ORDER]: [''],
 });
```

```
this.punchoutFormGroup.controls[FORM_CONTROL_NAME.ORDER].valueChanges
```

Just an idea.

Reply:
good idea! I'll move hardcoded value in a const. thx!